srcCxx/ShapeType_main.cxx

```diff
@@ -79,17 +132,22 @@ Topic *create_topic(DomainParticipant *participant, const char *topic_name)
     Topic *topic = participant->create_topic(topic_name, type_name,
             TOPIC_QOS_DEFAULT, NULL /* listener */, STATUS_MASK_NONE);
     if ( topic == NULL ) {
-        printf("create_topic error for topic '%s'\n", topic_name);
+        logger.log_message("failed to create topic: " + std::string(topic_name), Verbosity::ERROR);
         return NULL;
     }
 
     return topic;
 }
 
-ShapeTypeDataWriter *create_writer(DomainParticipant *participant, Topic *topic,
-                                   const char * partition, float livelinessPeriod)
+/*************************************************************/
+ShapeTypeDataWriter *create_writer(DomainParticipant *participant,
+                                   Topic *topic,
+                                   const char * partition,
+                                   float livelinessPeriod,
+                                   const char * color )
```

Comment:
It would be nice to have a consistent style for the where the `*` is placed (along with other style issues) since that makes reading the code easier.  Certainly not a must-have and I don't think anyone's attempted a style guide for these repos.

FWIW most modern C++ codebases I use would write these as `const char*` but of course there are tons of variations and everyone tends to think theirs is the "right" way.

Reply:
I thought about that, but I didn't want to just silently ignore it... 
Then I thought about printing an error and exiting, so test results wouldn't be misleading.  But that seemed about as useful as just leaving it as breaking change.

Reply:
updated for consistency